src/language-markdown/printer-markdown.js

```diff
@@ -52,6 +56,43 @@ const SIBLING_NODE_TYPES = new Set([
   "definition",
   "footnoteDefinition",
 ]);
+// https://en.wikipedia.org/wiki/Line_breaking_rules_in_East_Asian_languages
+/**
+ * The set of characters that must not immediately precede a line break
+ *
+ * e.g. `"（"`
+ *
+ * - Bad:  `"檜原村（\nひのはらむら）"`
+ * - Good: `"檜原村\n（ひのはらむら）"` or ``"檜原村（ひ\nのはらむら）"`
+ */
+const noBreakAfterSymbolSet = new Set(
```

Comment:
The word "symbol" has its own separate meaning in JS, so let's use "character" or "char" instead.

Reply:
Could you tell me whether not plural (“characters”)?

Reply:
I got it. I have already committed with singular.